### PR TITLE
[Relax] Fix batch normalization computation logic

### PR DESCRIPTION
**Pull request by Asuka0630**
Dear reviewers,

**Why**
The previous implementation of batch_norm incorrectly conflated the computation of mean and variance between training and evaluation modes. Additionally, for '_native_batch_norm_legit.no_stats‘, using instance_norm to handle normalization ignored the batch dimension, leading to incorrect behavior.

**How**
This PR includes the following fixes:
1. Corrects the computation logic to properly distinguish between training and evaluation modes.
2. Ensures the batch dimension is properly accounted for in `_batch_norm_legit_no_stats`.

**Environment**
GPU: NVIDIA A100-SXM4-80GB

### CODE REVIEW

File: python/tvm/topi/nn/batch_norm.py [hunk 1]
```diff
@@ -111,26 +111,22 @@ def batch_norm(
...
+        data_var = (
+            topi.sum((data - data_mean_rs) * (data - data_mean_rs), axis=reduce_axes) / shape_prod
+        )
+        data_var_rs = topi.reshape(data_var, shape)
+        out = (data - data_mean_rs) / topi.math.sqrt(data_var_rs + epsilon)
+    else:
         moving_mean_rs = topi.reshape(moving_mean, shape)
         moving_var_rs = topi.reshape(moving_var, shape)
-
         out = (data - moving_mean_rs) / topi.math.sqrt(moving_var_rs + epsilon)
```
[the remark this answers is not in the document]
Comment (tlopex) [reply python/tvm/topi/nn/batch_norm.py:130]: I think this review make sense, could you have a update?

File: python/tvm/relax/frontend/torch/exported_program_translator.py [hunk 2]
```diff
@@ -199,7 +199,7 @@ def _batch_norm_legit_no_stats(self, node: fx.Node) -> relax.Var:
 
         # Determine axes for instance norm (all spatial dimensions after channel)
         dim = len(self.shape_of(x))
-        axes = list(range(2, dim))
+        axes = [0] + list(range(2, dim))
```
[the remark this answers is not in the document]
Comment (tlopex) [reply python/tvm/relax/frontend/torch/exported_program_translator.py]: Could you update the comment?

File: tests/python/relax/test_transform_legalize_ops_nn.py [hunk 3]
```diff
@@ -2091,6 +2110,62 @@ def batch_norm(var_x: T.handle, var_gamma: T.handle, var_beta: T.handle, var_mov
...
+                                    T.writes(T_multiply_red[v_ax0])
+                                    with T.init():
+                                        T_multiply_red[v_ax0] = T.float32(0.0)
+                                    T_multiply_red[v_ax0] = T_multiply_red[v_ax0] + T_multiply[v_k0, v_ax0, v_k2, v_k3]
+                for ax0 in range(T.int64(3)):
+                    with T.block("T_divide_1"):
+                        v_ax0 = T.axis.spatial(T.int64(3), ax0)
+                        T.reads(T_multiply_red[v_ax0])
+                        T.writes(T_divide_1[v_ax0])
+                        T_divide_1[v_ax0] = T_multiply_red[v_ax0] / T.float32(1568.0)
```
Comment (tlopex) [inline tests/python/relax/test_transform_legalize_ops_nn.py:2168]: Could you briefly explain the modification you wrote for the test? I didn't quite catch it

File: python/tvm/relax/frontend/torch/exported_program_translator.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Asuka0630) [reply python/tvm/relax/frontend/torch/exported_program_translator.py]: ok, i will do that

File: tests/python/relax/test_transform_legalize_ops_nn.py (hunk 3 above)
Comment (Asuka0630) [reply tests/python/relax/test_transform_legalize_ops_nn.py:2168]: This change fixes the original erroneous structural check test for batch norm. The previous batch norm implementation was incorrect, so it couldn't match the TIR generated by the updated batch norm.

File: python/tvm/topi/nn/batch_norm.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Asuka0630) [reply python/tvm/topi/nn/batch_norm.py:130]: I don't think there will be any errors at runtime here, If I'm not misunderstanding Python. Then I need to add `data_mean = None` and `data_var = None` outside to avoid this warning. Is my understanding correct? ：)

File: tests/python/relax/test_transform_legalize_ops_nn.py (hunk 3 above)
Comment (tlopex) [reply tests/python/relax/test_transform_legalize_ops_nn.py:2168]: I see. Thank you for telling me about it
